tests/test_dse.py

```diff
@@ -1787,6 +1787,67 @@ def test_blocking_options(self, rotate):
         assert np.isclose(expected, norm(u1), rtol=1e-5)
         assert np.isclose(expected, norm(u2), rtol=1e-5)
 
+    def test_ftemps_option(self):
+        """
+        Test the compiler option `cire-ftemps=True`. This will make CIRE use
+        TempFunctions, rather than Arrays, to create temporaries, thus giving
+        control over allocation and deallocation to the user.
+        """
+        grid = Grid(shape=(30, 30, 30))
+        x, y, z = grid.dimensions
+        t = grid.stepping_dim
+
+        nthreads = 2
+        x0_blk0_size = 8
+        y0_blk0_size = 8
+
+        f = Function(name='f', grid=grid)
+        u = TimeFunction(name='u', grid=grid, space_order=3)
+        u1 = TimeFunction(name="u", grid=grid, space_order=3)
+        u2 = TimeFunction(name="u", grid=grid, space_order=3)
+
+        f.data_with_halo[:] = 1.
+        u.data_with_halo[:] = 0.32
+        u1.data_with_halo[:] = 0.32
+        u2.data_with_halo[:] = 0.32
+
+        eqn = Eq(u.forward, ((u[t, x, y, z] + u[t, x+1, y+1, z+1])*3*f +
+                             (u[t, x+2, y+2, z+2] + u[t, x+3, y+3, z+3])*3*f + 1))
+
+        op0 = Operator(eqn, opt=('noop', {'openmp': True}))
+        op1 = Operator(eqn, opt=('advanced', {'openmp': True, 'cire-mincost-sops': 1,
+                                              'cire-ftemps': True}))
+        op2 = Operator(eqn, opt=('advanced-fsg', {'openmp': True, 'cire-mincost-sops': 1,
+                                                  'cire-ftemps': True}))
+
+        op0(time_M=1, nthreads=nthreads)
+
+        # TempFunctions expect an override
+        try:
```

Comment:
Just saw this when looking for this example code. I think pytest gives you a mechanism for asserting that exceptions are thrown (and indeed, as I look again I see that this code _won't_ currently error in the case that no exception is thrown)

```
with pytest.raises(InvalidArgument):
    op1(...)
```

Reply:
ah, good to know! Will update the test.

> as I look again I see that this code won't currently error in the case that no exception is thrown)

That's my mistake. I should have added an exception below `op1(...)`